docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+- `directPeers` includes the exporter peer ID, to ensure it gets all messages
+- `subscriptionFilter` was injected to ensure a peer will connect to relevant topics, see [SubscriptionFilter](https://github.com/libp2p/go-libp2p-pubsub/blob/master/subscription_filter.go) interface
+- (fork `v1`) `msgID` is a custom function that calculates a `msg-id` based on the message content hash. 
+The default function uses the `sender` + `msg_seq` which we don't track, and enforces signature / verification for each message. 
+As all the messages are being verified using the share key, it would be redundant to it also in the pubusb level.
+Moreover, the default `msg-id` duplicates messages, causing it to be processed more than once, in case it was sent by multiple peers (e.g. decided message).
+- (fork `v1`) `signPolicy` was set to `StrictNoSign` (required for custom `msg-id`) to avoid producing and verifying message signatures in the pubsub router
+  - `signID` was set to empty (no author)
+
+### Pubsub Scoring
+
+`gossipsub v1.1` introduced pubsub [scoring](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#peer-scoring),
+the idea is that each individual peer maintains a score for other peers. 
+The score is locally computed by each individual peer based on observed behaviour and is not shared.
+
+`SSV.network` injects application specific scoring to apply connection and message scoring as part of pubsub scoring system. \
+See [Connection Scoring](#connection-scoring) and [Message Scoring](#message-scoring) for more information.
+
+Score thresholds are used by libp2p to determine whether a peer should be removed from topic's mesh, penalized or even ignored if the score drops too low. \
+See [this section](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#score-thresholds) for more details regards the different thresholds. \
+Thresholds values **TBD**, this section will be updated once that work is complete:
+
+- `gossipThreshold`: -4000
+- `publishThreshold`: -8000
+- `graylistThreshold`: -16000
+- `acceptPXThreshold`: 100
+- `opportunisticGraftThreshold`: 5
+
+[Score function](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#the-score-function) is running pariodically and will determine the score of peers. During heartbeat, the score it checked and bad peers are handled accordingly.
+
+
+### Message Scoring
+
+Message scorers track on operators' behavior w.r.t incoming IBFT messages:
+
+- Invalid message signature (`-100`)
+- Message from operator w/o shared committees (`-1000`)
+
+### Connection Scoring
+
+Peer's connection score is determined after a successful handshake,
+and peers with low score will be pruned.
+
+Connection scores are based on the following properties:
+
+- Shared subnets / committees (`25`)
+- Static nodes such as `exporter` (`10000`)
```

Comment:
Not sure this can be part of scoring as a malicious peer can simply say its an exporter when its not